components/proxy/src/main/java/com/hotels/styx/admin/handlers/UrlPatternRouter.java

```diff
@@ -140,15 +146,14 @@ public HttpHandler handler() {
             return placeholderNames;
         }
 
-        private Pattern compilePattern(String pattern) {
-            Pattern x = Pattern.compile(":([a-zA-Z0-9-_]+)");
-            Matcher matcher = x.matcher(pattern);
+        private static Pattern compilePattern(String pattern) {
+            Matcher matcher = PLACEHOLDER_PATTERN.matcher(pattern);
+
             return Pattern.compile(matcher.replaceAll("(?<$1>[a-zA-Z0-9-_]+)"));
```

Comment:
Should we use a name that depicts that we are replacing placeholders here for its appropriate regex? We could also completely  abstract the usage of 'standard Java regex', but I guess that's a bigger change.
Are we documenting the semantics of the UrlPatternRouter anywhere?

Reply:
I'm not documenting this yet. I think it is better to iron out any issues before providing a detailed documentation.